compiler/rustc_passes/src/loops.rs

```diff
@@ -227,8 +225,131 @@ impl<'a, 'hir> CheckLoopVisitor<'a, 'hir> {
             Closure(closure_span) => {
                 self.sess.dcx().emit_err(BreakInsideClosure { span, closure_span, name });
             }
-            AsyncClosure(closure_span) => {
-                self.sess.dcx().emit_err(BreakInsideAsyncBlock { span, closure_span, name });
+            Coroutine {
+                coroutine_span,
+                kind: hir::CoroutineDesugaring::Async,
+                source: hir::CoroutineSource::Block,
+            } => {
```

Comment:
You can make this into a single match statement and do:

```
let source = match source {
  ...
};
let kind = match kind {
  ...
};
```

I also wonder if we should just get rid of the article "a"/"an" from the error message so that we don't need to put that into the fluent file... Seems kind of annoying to have to put that in there -- it's definitely not very translatable. 🤔

Reply:
Yeah, same. The `of` followed by the article makes it wordy. 

Can I remove both `of` and the article to make it like ```cannot `break` inside `async` function``` in all places? It's concise and flows well.

Reply:
I guess that's fine, yes